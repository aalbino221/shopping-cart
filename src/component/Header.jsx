import React from 'react';

export default function Header() {
  return (
    <nav>
      <h1>PageByPage</h1>
      <ul>
        <li>Home</li>
        <li>Shop</li>
        <li>Contact</li>
        <li>
          <i className="fa-solid fa-cart-shopping" />
        </li>
      </ul>
    </nav>
  );
}
