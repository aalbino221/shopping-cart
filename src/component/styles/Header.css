.header {
  background-color: black;
  color: white;
  display: flex;
  width: 100%;
  justify-content: space-between;
  padding: 1.2rem 6rem;
  font-weight: bold;
  font-size: 1.4rem;
}

.header > ul {
  margin: 0;
  list-style-type: none;
  display: flex;
  gap: 4rem;
  align-items: center;
}

.header > ul > li {
  text-decoration: none;
}

.routerLink {
  color: white;
  text-decoration: none;
}
