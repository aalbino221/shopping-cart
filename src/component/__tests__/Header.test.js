import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'; // optional
import userEvent from '@testing-library/user-event';
import Header from '../Header';

describe('Header Component', () => {
  it('renders correct heading', () => {
    const { container } = render(<Header />);
    expect(screen.getByRole('heading')).toBeInTheDocument;
    expect(screen.getAllByRole('listitem')).toHaveLength(4);
    expect(container).toMatchSnapshot();
  });
});
