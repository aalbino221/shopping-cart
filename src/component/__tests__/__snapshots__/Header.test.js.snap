// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correct heading 1`] = `
<div>
  <nav>
    <h1>
      PageByPage
    </h1>
    <ul>
      <li>
        Home
      </li>
      <li>
        Shop
      </li>
      <li>
        Contact
      </li>
      <li>
        <i
          class="fa-solid fa-cart-shopping"
        />
      </li>
    </ul>
  </nav>
</div>
`;
